internal/cephfs/volume.go

```diff
@@ -42,11 +42,13 @@ const (
 type Subvolume struct {
 	BytesQuota    int      `json:"bytes_quota"`
 	DataPool      string   `json:"data_pool"`
+	Features      []string `json:"features"`
 	GID           int      `json:"gid"`
 	Mode          int      `json:"mode"`
 	MonAddrs      []string `json:"mon_addrs"`
 	Path          string   `json:"path"`
 	PoolNamespace string   `json:"pool_namespace"`
+	State         string   `json:"state"`
```

Comment:
what is this state and where it's getting used in this PR?

Reply:
The subvolume "state" is based on the current state of the subvolume and is also returned now with a subvolume info call. It contains one of the following values:
* "complete": subvolume is ready for all operations
* "snapshot-retained": subvolume is removed but its snapshots are retained

`State` got added as one of the parameters added in the [subvolume](https://github.com/ceph/ceph/blob/master/doc/cephfs/fs-volumes.rst#fs-subvolumes). As soon as PurgeVolume operation is performed retaining the snapshots, the state will change from `complete` to `snapshot-retained`.

The `Subvolume Info` calls also return the same to inform whether certain operations on a subvolume can be done.
For example, If a subvolume is in `snapshot-retained` state, `rm` operation will be disallowed on it.

Reply:
Are you doing any operation on this state in this PR?

Reply:
No Madhu, Just added it to keep csi subvolume parameters sync with ceph. Should I remove it for now and add later when used?

Reply:
Let it be there. just asked to make sure that am not missing anything when reviewing

Reply:
Thanks :)

Reply:
> The `Subvolume Info` calls also return the same to inform whether certain operations on a subvolume can be done.
> For example, If a subvolume is in `snapshot-retained` state, `rm` operation will be disallowed on it

Subvolume rm will not fail (it will be a No-OP), if removed again and state is "snapshot-retained" (assuming of course rm is with `--retain-snapshots` option). Other than that, storing the state is fine, can be leveraged as needed later.